assets/app/scripts/services/data.js

```diff
@@ -270,22 +276,41 @@ angular.module('openshiftConsole')
         return;
       }
 
-      self.create(resource, null, object, context, opts).then(
-        function (data) {
-          // include the original object, so the error handler can display the kind/name
-          data.object = object;
-          successResults.push(data);
-          remaining--;
-          _checkDone();
-        },
-        function (data) {
-          // include the original object, so the handler can display the kind/name
-          data.object = object;
-          failureResults.push(data);
-          remaining--;
-          _checkDone();
-        }
-      );
+      if (action === "update") {
+        self.update(resource, object.metadata.name, object, context, opts).then(
+          function (data) {
+            // include the original object, so the error handler can display the kind/name
+            data.object = object;
+            successResults.push(data);
+            remaining--;
+            _checkDone();
+          },
+          function (data) {
+            // include the original object, so the handler can display the kind/name
+            data.object = object;
+            failureResults.push(data);
+            remaining--;
+            _checkDone();
+          }
+        );
+      } else {
+        self.create(resource, null, object, context, opts).then(
+          function (data) {
+            // include the original object, so the error handler can display the kind/name
+            data.object = object;
+            successResults.push(data);
+            remaining--;
+            _checkDone();
+          },
+          function (data) {
+            // include the original object, so the handler can display the kind/name
+            data.object = object;
+            failureResults.push(data);
+            remaining--;
+            _checkDone();
+          }
+        );
```

Comment:
Since the callbacks are the same, make them named functions that you can reuse.

```javascript
var success = function(data) {
 // ...
};
var failure = function(data) {
 // ...
};

switch(action) {
case "update":
  self.update(resource, object.metadata.name, object, context, opts).then(success, failure);
  break;
default:
  self.create(resource, null, object, context, opts).then(success, failure);
  break;
}
```